### PR TITLE
fix: #79 if window does not exist, silently continue

### PR DESCRIPTION
**Pull request by JohnPhoto**

### CODE REVIEW

**Comment by ttmarek** [comment]
This is awesome @JohnPhoto. Thanks for taking the time to do this.

Just a couple of things:

The two unit tests you wrote are being run in a jsdom environment where `window` is always defined. So they don't really work.

The new version of Jest has a new feature where you can [define the test environment within a docblock](http://facebook.github.io/jest/docs/configuration.html#testenvironment-string). You can make two new files to hold your tests, something like `test/targets/google-analytics-node-env.js` and `test/targets/google-tag-manager-node-env.js`. And at the top of each file add a docblock to set the test environment. E.g.

```js
/**
 * @jest-environment node
 */

describe('If window does not exist', () => {
    it('should just silently not send events', () => {
        const events = [{ hitType: 'pageview' }];
        expect(() => GoogleAnalytics(events)).not.toThrow();
    });
});
```

Be sure to fetch upstream (this repo), and reinstall the dependencies to get the latest version of Jest.

One other thing, it's fine to fail silently when `window` is `undefined`, but it could be damaging to fail silently if `window.ga` or `window.dataLayer` is `undefined`. For instance, consider the scenario where a dev has set up redux beacon but has forgotten to add the Google Analytics snippet to their site. If we fail silently when `window.ga` is `undefined` then the dev would have no idea what they did wrong or what they forgot to do. I think it would actually be beneficial to add this to each target's spec. Something like:

```js
describe('If window.ga is undefined', () => {
    it('throws an error', () => {
        const events = [{ hitType: 'pageview' }];
        expect(() => GoogleAnalytics(events)).toThrow();
    });
});
```

To summarise: 
  - unit tests should be in a node environment (they are currently in a jsdom env)
  - shouldn't fail silently when `window.ga` or `window.dataLayer` is undefined.

**Comment by JohnPhoto** [comment]
Valid input, will adjust the PR

**Comment by JohnPhoto** [comment]
Updated the PR with requested changes.

**Comment by ttmarek** [comment]
LGTM. You're a champ @JohnPhoto! :1st_place_medal: :trophy: 
Thanks a lot eh.
